### DISCUSSION TITLE
Italics conversion... from an old school way, to today

### DISCUSSION THREAD

**Discussion by Charvelx04** [SubtitleEdit/subtitleedit#6497] [Ideas]
Okay... so far, I have this set up, having to go line by line... to manually put italics around the lines beginning with (') and ending with ('). This is a different method I see some companies used for offline voices and such, where we now put these in italics? I then, use Multiple Replace using < i >' replace with < i >, '< / i > with < i >... etc.

Question. Would it be of any value, to add say, an additional dialog window from the pull down menu, that would be a search feature, but only looks for certain symbols or letters at the beginning of each line. Just the beginning? It doesn't have to include the replacement code, as doing it manually is not difficult.

![Untitled](https://user-images.githubusercontent.com/35577096/206338815-1ee1ff5f-ff40-45c5-9649-c74393054fac.jpg)

**Comment by niksedk** [comment]
Why not use `Multiple replace` with a regular expression like this:
![image](https://user-images.githubusercontent.com/1261331/206914465-962d2368-53c1-4bcb-9e76-68e637824a72.png)

The first regex fixes single lines.
The second regex fixes two-liners.

**Comment by Charvelx04** [comment]
So, keeping the wildcard in the brackets, will that prevent losing words like: isn't, ain't, didn't, couldn't/wouldn't... etc, from changing into italics?

I was working with your idea, but lacked using the wildcard, and was replacing them manually.

![Untitled](https://user-images.githubusercontent.com/35577096/206926964-25bd6653-efee-4b83-9432-f44a4a26d0c6.jpg)

**Comment by niksedk** [comment]
The `()` is a grouping which can be used in the replace expression, the first will be `$1`, the second  `$2`
See https://stackoverflow.com/questions/3954927/how-to-replace-captured-groups-only

**Comment by Charvelx04** [comment]
Thank you. I have implemented those new replacements... and not sure why I typed in brackets so fast, when they are parentheses. 

I'll give that a read thru... appreciate it.

**Comment by Charvelx04** [comment]
Okay... hate to bring this back up, but need to know where I went wrong.

Got another sub to edit, only this time... we see two types. Sentences that begin and end with the apostrophe (')... and some sentences that just begin with it. Now, I would have assumed the Multiple Replace wouldn't recognize just the single instance, but she doesn't recognize either (screen1.jpg). Only when I manually turn them into italics, and use the additives I put in place, does it recognize the replace request (screen2.jpg). What am I doing wrong?


![screen1](https://user-images.githubusercontent.com/35577096/210157083-9a9b9510-09e1-4cc6-b10b-c439ee00d010.jpg)
![screen2](https://user-images.githubusercontent.com/35577096/210157084-c32baac7-3a74-4e68-8708-9a0657edb728.jpg)

**Reply by niksedk** [reply]
Could you attach the sub?

**Comment by Charvelx04** [comment]
Of course.
[Dawn of the Dead.zip](https://github.com/SubtitleEdit/subtitleedit/files/10329319/Dawn.of.the.Dead.zip)

**Reply by niksedk** [reply]
> Of course. [Dawn of the Dead.zip](https://github.com/SubtitleEdit/subtitleedit/files/10329319/Dawn.of.the.Dead.zip)

Some of the lines already has the italic tag...

**Comment by Charvelx04** [comment]
That's because SE auto saves my changes as fast as it can... here's the original, I removed the italics I added to show you the differences in what we are trying to achieve.
[Dawn of the Dead.zip](https://github.com/SubtitleEdit/subtitleedit/files/10333836/Dawn.of.the.Dead.zip)

**Reply by niksedk** [reply]
OK, that requires more rules:
![image](https://user-images.githubusercontent.com/1261331/210421167-011ca0f5-e23d-4bdc-b54f-917d1be8c7f5.png)

[replace-rules.zip](https://github.com/SubtitleEdit/subtitleedit/files/10339008/replace-rules.zip)

**Comment by Charvelx04** [comment]
Sweet... I'll add those rules here ASAP.

Thank you... I know it was a minor inconvenience to do it all manually, so I really do appreciate you helping resolve this. :)

**Comment by Charvelx04** [comment]
Okay... I think I'll have to manage it closely, although the expressions do what they are supposed to do, they really do what they are supposed to do. It solves one situation, but creates a new one. Easy to watch for, and to uncheck needed boxes.

Again.. thank you for your time.

![Untitled](https://user-images.githubusercontent.com/35577096/210438304-41646f32-4189-4758-abde-c44dc6374481.jpg)
